### PR TITLE
Convert functions docs to markdown

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Verified that output HTML is identical, has changes like wrapping with now visual change or are a visual improvement.

Note that the footnotes in the MySQl date functions section were inlined into the related table cells and last sentence in comparison.rst was using wrong pattern.. that is fixed in the md file.

## Additional context and related issues

Follow up to 

https://github.com/trinodb/trino/pull/17732
https://github.com/trinodb/trino/pull/17742
https://github.com/trinodb/trino/pull/17759

In parallel to https://github.com/trinodb/trino/pull/18375

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Review by ebyhr — CHANGES_REQUESTED** [review]
Some pages look a little broken. e.g. 
Screenshot 2023-07-25 at 11 32 53

**Comment by mosabua** [comment]
> Some pages look a little broken. For instance, the leading `\` should be removed. Screenshot 2023-07-25 at 11 32 53

yeah .. I found that and some others in my final check round .. will go over it all once more and push

**Comment by ebyhr** [comment]
> Verified that output HTML is identical, has changes like wrapping with now visual change or are a visual improvment.

Could you share the diff file so reviewers can confirm that?

**Comment by mosabua** [comment]
> > Verified that output HTML is identical, has changes like wrapping with now visual change or are a visual improvement.
> 
> Could you share the diff file so reviewers can confirm that?

I use a the "Compare folder" extension in Visual Code .. there are LOTS of differences in terms of wrapping, additional unused css classes and such. I am not sure it is worth your time doing that. Where should I share the diff files? Just here in the PR?

**Comment by ebyhr** [comment]
You can attach the archived file to this PR.

File: docs/src/main/sphinx/functions/comparison.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+```
+SELECT 'South_America' LIKE 'South\_America' ESCAPE '\';
+```
+
+The above query returns `true` since the escaped underscore symbol matches. If
+you need to match the used escape character as well, you can escape it.
+
+If you want to match for the chosen escape character, you simply escape itself.
+For example, you can use `\\` to match for `\`.
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/comparison.md:258]: This was wrong in the original document.

**Comment by mosabua** [comment]
> You can attach the archived file to this PR.

```


$ diff -qrc ~/tmp/html/functions/ docs/target/html/functions

diff /Users/manfred/tmp/html/functions/comparison.html and docs/target/html/functions/comparison.html differ
diff /Users/manfred/tmp/html/functions/datetime.html and docs/target/html/functions/datetime.html differ
diff /Users/manfred/tmp/html/functions/json.html and docs/target/html/functions/json.html differ
diff /Users/manfred/tmp/html/functions/logical.html and docs/target/html/functions/logical.html differ
diff /Users/manfred/tmp/html/functions/math.html and docs/target/html/functions/math.html differ
diff /Users/manfred/tmp/html/functions/ml.html and docs/target/html/functions/ml.html differ
diff /Users/manfred/tmp/html/functions/regexp.html and docs/target/html/functions/regexp.html differ
diff /Users/manfred/tmp/html/functions/string.html and docs/target/html/functions/string.html differ
diff /Users/manfred/tmp/html/functions/table.html and docs/target/html/functions/table.html differ
diff /Users/manfred/tmp/html/functions/teradata.html and docs/target/html/functions/teradata.html differ


$ diff /Users/manfred/tmp/html/functions/comparison.html docs/target/html/functions/comparison.html
681,682c681
< <p>Note that the value, min, and max parameters to <code class="docutils literal notranslate"><span class="pre">BETWEEN</span></code> and <code class="docutils literal notranslate"><span class="pre">NOT</span>
< <span class="pre">BETWEEN</span></code> must be the same type.  For example, Trino will produce an
---
> <p>Note that the value, min, and max parameters to <code class="docutils literal notranslate"><span class="pre">BETWEEN</span></code> and <code class="docutils literal notranslate"><span class="pre">NOT</span> <span class="pre">BETWEEN</span></code> must be the same type.  For example, Trino will produce an
717,724d715
< <colgroup>
< <col style="width: 13%"/>
< <col style="width: 13%"/>
< <col style="width: 15%"/>
< <col style="width: 15%"/>
< <col style="width: 19%"/>
< <col style="width: 26%"/>
< </colgroup>
880c871
< For example, you can use <code class="docutils literal notranslate"><span class="pre">\\</span></code> to match for ‘’'’.</p>
---
> For example, you can use <code class="docutils literal notranslate"><span class="pre">\\</span></code> to match for <code class="docutils literal notranslate"><span class="pre">\</span></code>.</p>


$ diff /Users/manfred/tmp/html/functions/datetime.html  docs/target/html/functions/datetime.html

623,627d622
< <colgroup>
< <col style="width: 9%"/>
< <col style="width: 60%"/>
< <col style="width: 31%"/>
< </colgroup>
914,917d908
< <colgroup>
< <col style="width: 29%"/>
< <col style="width: 71%"/>
< </colgroup>
971,974d961
< <colgroup>
< <col style="width: 49%"/>
< <col style="width: 51%"/>
< </colgroup>
1052,1055d1038
< <colgroup>
< <col style="width: 35%"/>
< <col style="width: 65%"/>
< </colgroup>
1124,1127d1106
< <colgroup>
< <col style="width: 7%"/>
< <col style="width: 93%"/>
< </colgroup>
1141c1120
< <td><p>Month, numeric (<code class="docutils literal notranslate"><span class="pre">1</span></code> .. <code class="docutils literal notranslate"><span class="pre">12</span></code>) <a class="footnote-reference brackets" href="#z" id="id1" role="doc-noteref"><span class="fn-bracket">[</span>4<span class="fn-bracket">]</span></a></p></td>
---
> <td><p>Month, numeric (<code class="docutils literal notranslate"><span class="pre">1</span></code> .. <code class="docutils literal notranslate"><span class="pre">12</span></code>), this specifier does not support <code class="docutils literal notranslate"><span class="pre">0</span></code> as a month.</p></td>
1147c1126
< <td><p>Day of the month, numeric (<code class="docutils literal notranslate"><span class="pre">01</span></code> .. <code class="docutils literal notranslate"><span class="pre">31</span></code>) <a class="footnote-reference brackets" href="#z" id="id2" role="doc-noteref"><span class="fn-bracket">[</span>4<span class="fn-bracket">]</span></a></p></td>
---
> <td><p>Day of the month, numeric (<code class="docutils literal notranslate"><span class="pre">01</span></code> .. <code class="docutils literal notranslate"><span class="pre">31</span></code>), this specifier does not support <code class="docutils literal notranslate"><span class="pre">0</span></code> as a month or day.</p></td>
1150c1129
< <td><p>Day of the month, numeric (<code class="docutils literal notranslate"><span class="pre">1</span></code> .. <code class="docutils literal notranslate"><span class="pre">31</span></code>) <a class="footnote-reference brackets" href="#z" id="id3" role="doc-noteref"><span class="fn-bracket">[</span>4<span class="fn-bracket">]</span></a></p></td>
---
> <td><p>Day of the month, numeric (<code class="docutils literal notranslate"><span class="pre">1</span></code> .. <code class="docutils literal notranslate"><span class="pre">31</span></code>), this specifier does not support <code class="docutils literal notranslate"><span class="pre">0</span></code> as a day.</p></td>
1153c1132
< <td><p>Fraction of second (6 digits for printing: <code class="docutils literal notranslate"><span class="pre">000000</span></code> .. <code class="docutils literal notranslate"><span class="pre">999000</span></code>; 1 - 9 digits for parsing: <code class="docutils literal notranslate"><span class="pre">0</span></code> .. <code class="docutils literal notranslate"><span class="pre">999999999</span></code>) <a class="footnote-reference brackets" href="#f" id="id4" role="doc-noteref"><span class="fn-bracket">[</span>1<span class="fn-bracket">]</span></a></p></td>
---
> <td><p>Fraction of second (6 digits for printing: <code class="docutils literal notranslate"><span class="pre">000000</span></code> .. <code class="docutils literal notranslate"><span class="pre">999000</span></code>; 1 - 9 digits for parsing: <code class="docutils literal notranslate"><span class="pre">0</span></code> .. <code class="docutils literal notranslate"><span class="pre">999999999</span></code>), timestamp is truncated to milliseconds.</p></td>
1180c1159
< <td><p>Month, numeric (<code class="docutils literal notranslate"><span class="pre">01</span></code> .. <code class="docutils literal notranslate"><span class="pre">12</span></code>) <a class="footnote-reference brackets" href="#z" id="id5" role="doc-noteref"><span class="fn-bracket">[</span>4<span class="fn-bracket">]</span></a></p></td>
---
> <td><p>Month, numeric (<code class="docutils literal notranslate"><span class="pre">01</span></code> .. <code class="docutils literal notranslate"><span class="pre">12</span></code>), this specifier does not support <code class="docutils literal notranslate"><span class="pre">0</span></code> as a month.</p></td>
1213c1192
< <td><p>Day of the week (<code class="docutils literal notranslate"><span class="pre">0</span></code> .. <code class="docutils literal notranslate"><span class="pre">6</span></code>), where Sunday is the first day of the week <a class="footnote-reference brackets" href="#w" id="id6" role="doc-noteref"><span class="fn-bracket">[</span>3<span class="fn-bracket">]</span></a></p></td>
---
> <td><p>Day of the week (<code class="docutils literal notranslate"><span class="pre">0</span></code> .. <code class="docutils literal notranslate"><span class="pre">6</span></code>), where Sunday is the first day of the week, this specifier is not supported,consider using <a class="reference internal" href="#day_of_week" title="day_of_week"><code class="xref py py-func docutils literal notranslate"><span class="pre">day_of_week()</span></code></a> (it uses <code class="docutils literal notranslate"><span class="pre">1-7</span></code> instead of <code class="docutils literal notranslate"><span class="pre">0-6</span></code>).</p></td>
1225c1204
< <td><p>Year, numeric (two digits) <a class="footnote-reference brackets" href="#y" id="id7" role="doc-noteref"><span class="fn-bracket">[</span>2<span class="fn-bracket">]</span></a></p></td>
---
> <td><p>Year, numeric (two digits), when parsing, two-digit year format assumes range <code class="docutils literal notranslate"><span class="pre">1970</span></code> .. <code class="docutils literal notranslate"><span class="pre">2069</span></code>, so “70” will result in year <code class="docutils literal notranslate"><span class="pre">1970</span></code> but “69” will produce <code class="docutils literal notranslate"><span class="pre">2069</span></code>.</p></td>
1235,1251d1213
< <aside class="footnote brackets" id="f" role="note">
< <span class="label"><span class="fn-bracket">[</span><a href="#id4" role="doc-backlink">1</a><span class="fn-bracket">]</span></span>
< <p>Timestamp is truncated to milliseconds.</p>
< </aside>
< <aside class="footnote brackets" id="y" role="note">
< <span class="label"><span class="fn-bracket">[</span><a href="#id7" role="doc-backlink">2</a><span class="fn-bracket">]</span></span>
< <p>When parsing, two-digit year format assumes range <code class="docutils literal notranslate"><span class="pre">1970</span></code> .. <code class="docutils literal notranslate"><span class="pre">2069</span></code>, so “70” will result in year <code class="docutils literal notranslate"><span class="pre">1970</span></code> but “69” will produce <code class="docutils literal notranslate"><span class="pre">2069</span></code>.</p>
< </aside>
< <aside class="footnote brackets" id="w" role="note">
< <span class="label"><span class="fn-bracket">[</span><a href="#id6" role="doc-backlink">3</a><span class="fn-bracket">]</span></span>
< <p>This specifier is not supported yet. Consider using <a class="reference internal" href="#day_of_week" title="day_of_week"><code class="xref py py-func docutils literal notranslate"><span class="pre">day_of_week()</span></code></a> (it uses <code class="docutils literal notranslate"><span class="pre">1-7</span></code> instead of <code class="docutils literal notranslate"><span class="pre">0-6</span></code>).</p>
< </aside>
< <aside class="footnote brackets" id="z" role="note">
< <span class="label"><span class="fn-bracket">[</span>4<span class="fn-bracket">]</span></span>
< <span class="backrefs">(<a href="#id1" role="doc-backlink">1</a>,<a href="#id2" role="doc-backlink">2</a>,<a href="#id3" role="doc-backlink">3</a>,<a href="#id5" role="doc-backlink">4</a>)</span>
< <p>This specifier does not support <code class="docutils literal notranslate"><span class="pre">0</span></code> as a month or day.</p>
< </aside>
1294,1297d1255
< <colgroup>
< <col style="width: 45%"/>
< <col style="width: 55%"/>
< </colgroup>

$ diff /Users/manfred/tmp/html/functions/json.html  docs/target/html/functions/json.html 

1409,1412d1408
< <colgroup>
< <col style="width: 16%"/>
< <col style="width: 84%"/>
< </colgroup>
1441,1444d1436
< <colgroup>
< <col style="width: 33%"/>
< <col style="width: 67%"/>
< </colgroup>
1464,1465c1456
< not have three or more children. This error is handled according to the <code class="docutils literal notranslate"><span class="pre">ON</span>
< <span class="pre">ERROR</span></code> clause.</p>
---
> not have three or more children. This error is handled according to the <code class="docutils literal notranslate"><span class="pre">ON</span> <span class="pre">ERROR</span></code> clause.</p>
1477,1480d1467
< <colgroup>
< <col style="width: 36%"/>
< <col style="width: 64%"/>
< </colgroup>
1555,1556c1542
< in a JSON array. <code class="docutils literal notranslate"><span class="pre">WITHOUT</span> <span class="pre">ARRAY</span> <span class="pre">WRAPPER</span></code> is the default option. <code class="docutils literal notranslate"><span class="pre">WITH</span>
< <span class="pre">CONDITIONAL</span> <span class="pre">ARRAY</span> <span class="pre">WRAPPER</span></code> wraps every result which is not a singleton JSON
---
> in a JSON array. <code class="docutils literal notranslate"><span class="pre">WITHOUT</span> <span class="pre">ARRAY</span> <span class="pre">WRAPPER</span></code> is the default option. <code class="docutils literal notranslate"><span class="pre">WITH</span> <span class="pre">CONDITIONAL</span> <span class="pre">ARRAY</span> <span class="pre">WRAPPER</span></code> wraps every result which is not a singleton JSON
1565,1568d1550
< <colgroup>
< <col style="width: 16%"/>
< <col style="width: 84%"/>
< </colgroup>
1597,1600d1578
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>
1634,1637d1611
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>
1668,1671d1641
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>
1706,1709d1675
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>
1727,1728c1693
< <p>The following query shows the result of the <code class="docutils literal notranslate"><span class="pre">QUOTES</span></code> clause. Note that <code class="docutils literal notranslate"><span class="pre">KEEP</span>
< <span class="pre">QUOTES</span></code> is the default.</p>
---
> <p>The following query shows the result of the <code class="docutils literal notranslate"><span class="pre">QUOTES</span></code> clause. Note that <code class="docutils literal notranslate"><span class="pre">KEEP</span> <span class="pre">QUOTES</span></code> is the default.</p>
1737,1741d1701
< <colgroup>
< <col style="width: 24%"/>
< <col style="width: 38%"/>
< <col style="width: 38%"/>
< </colgroup>
1844,1847d1803
< <colgroup>
< <col style="width: 16%"/>
< <col style="width: 84%"/>
< </colgroup>
1875,1878d1830
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>
1907,1910d1858
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>
1940,1943d1887
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>
1973,1976d1916
< <colgroup>
< <col style="width: 38%"/>
< <col style="width: 62%"/>
< </colgroup>

$ diff /Users/manfred/tmp/html/functions/logical.html  docs/target/html/functions/logical.html 
590,594d589
< <colgroup>
< <col style="width: 19%"/>
< <col style="width: 65%"/>
< <col style="width: 16%"/>
< </colgroup>
642,647d636
< <colgroup>
< <col style="width: 25%"/>
< <col style="width: 25%"/>
< <col style="width: 25%"/>
< <col style="width: 25%"/>
< </colgroup>
709,712d697
< <colgroup>
< <col style="width: 50%"/>
< <col style="width: 50%"/>
< </colgroup>


$ diff /Users/manfred/tmp/html/functions/math.html  docs/target/html/functions/math.html 

614,617d613
< <colgroup>
< <col style="width: 15%"/>
< <col style="width: 85%"/>
< </colgroup>

$ diff /Users/manfred/tmp/html/functions/ml.html  docs/target/html/functions/ml.html 

399c399
<         <li class="md-nav__item"><a href="#id2" class="md-nav__link">Machine learning functions</a>
---
>         <li class="md-nav__item"><a href="#id1" class="md-nav__link">Machine learning functions</a>
582c582
<         <li class="md-nav__item"><a href="#id2" class="md-nav__link">Machine learning functions</a>
---
>         <li class="md-nav__item"><a href="#id1" class="md-nav__link">Machine learning functions</a>
702,703c702,703
< <section id="id2">
< <h2 id="id2">Machine learning functions<a class="headerlink" href="#id2" title="Permalink to this heading">#</a></h2>
---
> <section id="id1">
> <h2 id="id1">Machine learning functions<a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h2>



$ diff /Users/manfred/tmp/html/functions/regexp.html  docs/target/html/functions/regexp.html 

608,609c608
< Example: <code class="docutils literal notranslate"><span class="pre">\p{NoncharacterCodePoint}</span></code></p>
< </li>
---
> Example: <code class="docutils literal notranslate"><span class="pre">\p{NoncharacterCodePoint}</span></code></p></li>

$ diff /Users/manfred/tmp/html/functions/string.html  docs/target/html/functions/string.html 

750c750
< <dd><dl>
---
> <dd><dl class="simple myst">
911,914d910
< <colgroup>
< <col style="width: 11%"/>
< <col style="width: 89%"/>
< </colgroup>

$ diff /Users/manfred/tmp/html/functions/table.html  docs/target/html/functions/table.html 

614,615c614
< <p>For the list of built-in table functions available in Trino, see <a class="reference internal" href="#built-in-table-functions"><span class="std std-ref">built in
< table functions</span></a>.</p>
---
> <p>For the list of built-in table functions available in Trino, see <a class="reference internal" href="#built-in-table-functions"><span class="std std-ref">built in table functions</span></a>.</p>
620,621c619
< more information about supported table functions, refer to the <a class="reference internal" href="../connector.html"><span class="doc">connector
< documentation</span></a>.</p>
---
> more information about supported table functions, refer to the <a class="reference internal" href="../connector.html"><span class="doc">connector documentation</span></a>.</p>

$ diff /Users/manfred/tmp/html/functions/teradata.html  docs/target/html/functions/teradata.html 

607,610d606
< <colgroup>
< <col style="width: 31%"/>
< <col style="width: 69%"/>
< </colgroup>
```

**Comment by ebyhr** [comment]
"MySQL date functions" is different from the original table. 

Before 
Screenshot 2023-07-26 at 8 45 15

After
Screenshot 2023-07-26 at 8 45 30

**Comment by ebyhr** [comment]
I see you modified PR description.

**Comment by mosabua** [comment]
> > Note that the footnotes in the MySQl date functions section were inlined into the related table cells
> 
> Could you share the reason why you inlined?

The markdown render put the footnotes at the bottom of the page and that made things worse. I tried to move it back up but found that even in the original form the footnotes at the bottom of the table as just very much out of context. Since there was not much text anyway moving it into the table made everything better for the reader.

**Review by electrum — APPROVED** [review]

**Pull request merged**